core/src/com/unciv/logic/civilization/diplomacy/CityStateFunctions.kt

```diff
@@ -677,7 +677,7 @@ class CityStateFunctions(val civInfo: Civilization) {
     fun getCityStateResourcesForAlly() = ResourceSupplyList().apply {
         for (city in civInfo.cities) {
             // IGNORE the fact that they consume their own resources - #4769
-            addPositiveByResource(city.getCityResources(), Constants.cityStates)
+            addPositiveByResource(city.getResourcesGeneratedByCity(), Constants.cityStates)
         }
```

Comment:
Why is this not civ resources? Mods can give city states resources to trade

Reply:
Good point!!

Reply:
I remember why we did this, it was to avoid the fact that resources are consumed - this means kind of rebuilding the getCivResources... I'll add a todo here, out of scope for this PR